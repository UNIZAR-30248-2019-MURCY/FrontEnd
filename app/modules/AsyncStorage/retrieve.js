import { AsyncStorage } from 'react-native';

export const retrieveItem = async (key) => {
    let data ='';
    try {
        data = await AsyncStorage.getItem(key)
        console.log(data)
    } catch (error) {
        console.log(error.message);
    }
    return data;
}
