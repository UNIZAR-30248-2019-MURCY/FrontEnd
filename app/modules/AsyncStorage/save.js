import { AsyncStorage } from 'react-native';

export const saveData = async (key, data) => {
    try {
        await AsyncStorage.setItem(key, JSON.stringify(data));
        console.log('data stored');
    } catch (error) {
        // Error retrieving data
        console.log(error.message);
    }
};
