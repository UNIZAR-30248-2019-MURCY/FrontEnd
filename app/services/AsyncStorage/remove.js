import { AsyncStorage } from 'react-native';

export const removeItem = async (key) => {
    try {
        await AsyncStorage.removeItem(key);
        console.log('data removed');
        return true;
    } catch (error) {
        console.log(error.message);
        return false;
    }
}
