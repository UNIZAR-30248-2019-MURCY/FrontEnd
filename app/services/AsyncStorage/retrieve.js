import { AsyncStorage } from 'react-native';

export const retrieveItem = async (key) => {
    let data ='';
    try {
        let dataStored = await AsyncStorage.getItem(key)
        data = JSON.parse(dataStored).token;
        console.log('retrievedItem')
    } catch (error) {
        console.log(error.message);
    }
    return data;
}
