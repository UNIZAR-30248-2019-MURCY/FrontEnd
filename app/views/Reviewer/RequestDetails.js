import React, {Component} from 'react';
import {AsyncStorage, FlatList, StyleSheet, TextInput, View,} from 'react-native'
import {Button, ListItem, Text} from 'react-native-elements';

export default class RequestDetails extends Component {

    constructor(props) {
        super(props);
        this.state = {
            workflow: false
        }
    }

    componentDidMount() {
        if (this.props.navigation) {
            this.setState({workflow: this.props.navigation.getParam('workflow', 'default value')});
        }
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.navigation.state.params.lastName) {
            this.setState({lastName: nextProps.navigation.state.params.lastName});
        }
    }

    render() {
        let showDetails = (
            this.state.workflow ?
                <View style={styles.containerRequest}>
                    <Text h4 style={styles.containerSubTitle}>{this.state.workflow.title}</Text>
                    <Text style={styles.containerRequestDetails}>
                        Description: {this.state.workflow.description}
                        {'\n'}{'\n'}
                        Status: {this.state.workflow.status}
                    </Text>
                    <View style={styles.containerButtons}>
                        <Button
                            buttonStyle={styles.button}
                            title="Accept"
                            onPress={() => {
                                this.props.navigation.replace('RequestActionConfirm', { type: 'accepted' });
                            }}/>

                        <Button
                            buttonStyle={styles.buttonDen}
                            title="Deny"
                            onPress={() => {
                                this.props.navigation.replace('RequestActionConfirm', { type: 'denied' });
                            }}/>
                    </View>
                </View> :
                <View></View>

        );

        return (
            <View style={styles.container}>
                <View style={styles.containerTitle}>
                    <Text h2>Details</Text>
                </View>
                <View style={styles.containerRequest2}>
                    {showDetails}
                    <View style={styles.containerRequest3}>
                        <Button
                            className='return-button'
                            type="clear"
                            buttonStyle={styles.button2}
                            title="Return"
                            titleStyle={{color: 'grey'}}
                            onPress={() => {
                                this.props.navigation.goBack();
                            }
                            }/>
                    </View>
                </View>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: 20,
        //justifyContent: 'center',
    },
    containerTitle: {
        alignItems: 'center',
        marginTop: 50,
        padding: 20
    },
    containerSubTitle: {
        marginTop: 25
    },
    containerRequest: {
        flex: 1,
        alignItems: 'center',
    },
    containerRequest2: {
        flex: 1,
    },
    containerRequest3: {
        alignItems: 'center',
    },
    containerRequestDetails: {
        flex: 1,
        marginTop: 25,
        fontSize: 18,
        color: 'grey',
        textAlign: 'center',
    },
    button: {
        width: 150,
        height: 55,
        marginTop: 20,
        margin: 10,
        backgroundColor: 'grey',
        borderRadius: 14
    },
    buttonDen: {
        width: 150,
        height: 55,
        marginTop: 20,
        margin: 10,
        backgroundColor: 'red',
        borderRadius: 14
    },
    button2: {
        marginBottom: 10,
        width: 150,
        height: 55,
        borderRadius: 14
    },
    containerButtons: {
        flexDirection: 'row'
    },

})
