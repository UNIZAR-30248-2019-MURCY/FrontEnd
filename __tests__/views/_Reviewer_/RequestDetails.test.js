import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import RequestDetails from "../../../app/views/Reviewer/RequestDetails";
import {retrieveItem} from "../../../app/services/AsyncStorage/retrieve";

describe('<RequestDetails />', () => {
    it('RequestDetails renders without crashing', () => {
        const page = renderer.create(<RequestDetails />).toJSON();
        expect(page).toBeTruthy();
    });
    it('RequestDetails test against snapshot', () => {
        const page = renderer.create(<RequestDetails />).toJSON();
        expect(page).toMatchSnapshot();
    });
});
