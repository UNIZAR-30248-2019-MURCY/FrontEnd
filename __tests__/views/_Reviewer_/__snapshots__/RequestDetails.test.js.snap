// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestDetails /> RequestDetails test against snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 50,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 42.5,
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Details
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "padding": 25,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 27.5,
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "flex": 1,
          "fontSize": 18,
          "marginTop": 25,
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Description: 
      

      

      Status: 
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 14,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "grey",
                "borderColor": "#2089dc",
                "borderRadius": 14,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 55,
                "justifyContent": "center",
                "margin": 10,
                "marginTop": 20,
                "padding": 8,
                "width": 150,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Accept
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderRadius": 14,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "grey",
                "borderColor": "#2089dc",
                "borderRadius": 14,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 55,
                "justifyContent": "center",
                "margin": 10,
                "marginTop": 20,
                "padding": 8,
                "width": 150,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Deny
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
